lib/pharos/phase.rb

```diff
@@ -14,8 +14,17 @@ def to_s
       "#{self.class.title} @ #{@host}"
     end
 
-    def self.register_component(component)
-      Pharos::Phases.register_component component
+    # Register a component to the Pharos::Phases component registry
+    # @param name [String]
+    # @param version [String]
+    # @param license [String]
+    def self.register_component(name, version:, license:)
+      slug = name.gsub(/[^a-z0-9]+/, '_')
```

Comment:
If the `name` is used as an openstruct method name, then should it be a symbol instead of a string? 

Alternatively just use `components['cri-o'].version` instead?

Reply:
Can't be a symbol beacuse the name is shown in the component version listing `pharos-cluster version`. Otherwise would have to do something like `register_component :foo, name: 'foo'`